### PR TITLE
SDI-568: ✨ Upgrade to spring boot 3

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: .trivyignore [hunk 1]
```diff
@@ -32,6 +32,3 @@ CVE-2022-42889
 # Suppression for h2 2.1.214 password on command line vulnerability
 #   can be suppressed as we only run h2 locally and not on build environments
 CVE-2022-45868
-# Suppression for spring-web 5.3.24 as bundled with spring boot
```
Comment (petergphillips) [inline .trivyignore:35]: we're using spring web 6 so isn't relevant (automatically suppressed by spring boot upgrade)

File: src/main/resources/application-localstack.yml [hunk 2]
```diff
@@ -3,27 +3,20 @@ hmpps.sqs:
   queues:
     prisoner:
       queueName: hmpps_prisoner_to_nomis_queue
-      queueArn: arn:aws:sns:eu-west-2:000000000000:hmpps_prisoner_to_nomis_queue
```
Comment (petergphillips) [inline src/main/resources/application-localstack.yml:6]: not a property of the queue anyway

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -60,7 +60,6 @@ spring:
       dev:
         - "localstack"
       test:
-        - "localstack"
```
Comment (petergphillips) [inline src/main/resources/application.yml:63]: remove so can override for tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt [hunk 4]
```diff
@@ -115,17 +112,6 @@ fun activityRetryMessage() = """
       }
 """.trimIndent()
 
-private object logger {
-  val log: Logger = LoggerFactory.getLogger(this::class.java)
-}
-
-fun getNumberOfMessagesCurrentlyOnQueue(awsSqsClient: AmazonSQS, url: String): Int? {
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt:122]: removed as part of the sqs library these days

File: src/test/resources/application-test.yml [hunk 5]
```diff
@@ -13,3 +13,37 @@ api.base.url:
   incentives: http://localhost:8085
   activities: http://localhost:8086
   sentence.adjustments: http://localhost:8087
+
+hmpps.sqs:
+  provider: localstack
+  queues:
+    prisoner:
+      queueName: ${random.uuid}
```
Comment (petergphillips) [inline src/test/resources/application-test.yml:21]: test settings that ensure that multiple runs don't interfere with each other

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesDomainEventListener.kt [hunk 6]
```diff
@@ -24,7 +24,7 @@ class ActivitiesDomainEventListener(
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  @JmsListener(destination = "activity", containerFactory = "hmppsQueueContainerFactoryProxy")
+  @SqsListener("activity", factory = "hmppsQueueContainerFactoryProxy", maxMessagesPerPoll = "1", maxInflightMessagesPerQueue = "1")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesDomainEventListener.kt:27]: This makes me nervous after the experience with offender-events-ui. It took a few days for the RejectedTaskExecution error to happen in that project... @andymarke Do you think we need to process 1 message at a time?

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**